docs/Recipes/faking-http-client.md

```diff
@@ -0,0 +1,98 @@
+
+Let's assume that you want to create a fake `HttpClient` so you can dictate the
+behavior of the
+[GetAsync(String)](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getasync?view=net-7.0#system-net-http-httpclient-getasync(system-string))
+method. This seems like it would be a
+straightforward task, but it's complicated by the design of `HttpClient`, which
+is not faking-friendly.
+
+## A working Fake
+
+First off, let's look at the declaration of `GetAsync`:
+
+```csharp
+public Task<HttpResponseMessage>
+    GetAsync([StringSyntax(StringSyntaxAttribute.Uri)] string? requestUri)
+```
+
+This method is neither virtual nor abstract, and so [can't be overridden by
+FakeItEasy](../what-can-be-faked.md#what-members-can-be-overridden).
+
+As a workaround, we can look at the [definition of
+GetAsync](https://github.com/dotnet/runtime/blob/ab5e28c1cab305450897749daa7393bef30d7505/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L363-L364)
+and see that the method eventually ends up calling
+[HttpMessageHandler.SendAsync(HttpRequestMessage,
+CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpmessagehandler.sendasync?view=net-7.0#system-net-http-httpmessagehandler-sendasync(system-net-http-httprequestmessage-system-threading-cancellationtoken))
+on an `HttpMessageHandler` that can be supplied via the `HttpClient`
+constructor.
+
+`HttpMessageHandler.SendAsync` is protected, which makes it
+less convenient to override than a public method. We need to specify the call by
+name, and to give FakeItEasy a hint about the return type, as described in
+[Specifying a call to any method or
+property](../specifying-a-call-to-configure.md#specifying-a-call-to-any-method-or-property).
+
+With this knowledge, we can write a passing test:
+
+??? note "This is a simplified example"
+    In the interest of brevity, we create a Fake, exercise it directly, and check
+    its behavior. A more realistic example would create the Fake as a collaborator
+    of some production class (the "system under test") and the Fake would not be
+    called directly from the test code.
+
+```csharp
+public async Task Test()
+{
+    var response = new HttpResponseMessage();
+    response.Content = new StringContent("FakeItEasy is fun");

```

Comment:
nit: maybe use object initializer syntax here?
```suggestion
    var response = new HttpResponseMessage
    {
        Content = new StringContent("FakeItEasy is fun")
    };

```

Reply:
I originally had it, but one one line, and it didn't render well. Will try your version.